### ISSUE TITLE
Extended structure of groups, keywords and urls

### ISSUE THREAD

**Issue by Andrii-D** [serphacker/serposcope#94]
First of all I want to say thanks to all creators, you did a great job. Especially thanks for dockerizing!
We have reviewed your system with our team and we would like to contribute to development process.

Here is a request:
Lets say I have 100k pages (unique urls) with 20 keywords per each page. So I will create 100k groups of keywords to be able to track position of every specific url. But it is going to be a very big mess with 100k groups and viewing statistic. So it would be nice to organize groups by some kind of sub-groups.

Suggested solution: Tag cloud!
Probably the best way of solving this problem is to have ability to create and add tags for every group (or maybe for every keyword). With this kind of system you can create any structure you want for groups of keywords and then view statistic by searching by tags.

What do you think about this feature? 
Do you have any other solutions?
What do you think is the best way to implement it?

**Comment by serphacker** [comment]
Yes, the rendering is bit messy when you have a lot of keywords, groups have their limitation.

I think group will be reworked, and just dedicated to access control (ie restricted set of keywords for customers). 

And then I will introduce tags per keyword, I have to think about it.
